R/arg.R

```diff
@@ -52,6 +53,14 @@ arg_match <- function(arg, values = NULL) {
     i_partial <- pmatch(arg, values)
     if (!is_na(i_partial)) {
       candidate <- values[[i_partial]]
+    }
+
+    i_close <- adist(arg, values) / nchar(values)
```

Comment:
I'd like to see a few more tests on short strings — I'd like to see how this deals with false positives too.

Reply:
I've added a few more, as I was running them it looked (to me) like my original threshold of 0.3 was too strict for short strings (though 0.5 might be too lenient for longer ones).

Reply:
I'd wonder if just using a constant (e.g. edit distance of 3) might be better? Would you mind trying that out?

Reply:
Yeah, I just tried that out, and things get a bit funky with small strings:

```
> myarg <- "a"
> arg_match(myarg, c("b", "c"))
Error: `myarg` must be one of "b" or "c"
Did you mean "b"?
```

```
> myarg <- "zz"
> arg_match(myarg, c("aa", "dd"))
Error: `myarg` must be one of "aa" or "dd"
Did you mean "aa"?
```

We could also disable the typo-checking for short strings with a constant, but that might not be better than the ratio. At the risk of overcomplicating: a max distance + a ratio might be a strike a balance here (something like `any(constant <= 3 & ratio <= 0.5)`).

Reply:
Ok, lets stick with what you have and we can tweak it in the future, if needed.